Sources/SwiftParser/Parser.swift

```diff
@@ -173,27 +233,49 @@ public struct Parser {
     parseTransition: IncrementalParseTransition? = nil,
     arena: ParsingSyntaxArena? = nil
   ) {
-    self.maximumNestingLevel = maximumNestingLevel ?? Self.defaultMaximumNestingLevel
+    self.init(
+      _buffer: input,
+      maximumNestingLevel: maximumNestingLevel,
+      parseTransition: parseTransition,
+      arena: arena,
+      experimentalFeatures: []
+    )
+  }
 
-    var sourceBuffer: UnsafeBufferPointer<UInt8>
-    if let arena {
-      self.arena = arena
-      sourceBuffer = input
-      precondition(arena.contains(text: SyntaxText(baseAddress: input.baseAddress, count: input.count)))
-    } else {
-      self.arena = ParsingSyntaxArena(
-        parseTriviaFunction: TriviaParser.parseTrivia(_:position:)
-      )
-      sourceBuffer = self.arena.internSourceBuffer(input)
-    }
+  /// Initializes a ``Parser`` from the given input string, with a given set
+  /// of experimental language features.
+  @_spi(ExperimentalLanguageFeatures)
+  public init(
+    _ input: String,
+    maximumNestingLevel: Int? = nil,
+    parseTransition: IncrementalParseTransition? = nil,
+    experimentalFeatures: ExperimentalFeatures
+  ) {
+    self.init(
+      _string: input,
+      maximumNestingLevel: maximumNestingLevel,
+      parseTransition: parseTransition,
+      experimentalFeatures: experimentalFeatures
+    )
+  }
 
-    self.lexemes = Lexer.tokenize(sourceBuffer, lookaheadTracker: lookaheadTrackerOwner.lookaheadTracker)
-    self.currentToken = self.lexemes.advance()
-    if let parseTransition {
-      self.parseLookup = IncrementalParseLookup(transition: parseTransition)
-    } else {
-      self.parseLookup = nil
-    }
+  /// Initializes a ``Parser`` from the given input buffer, with a given set
+  /// of experimental language features.
+  @_spi(ExperimentalLanguageFeatures)
+  public init(
```

Comment:
I can't say I love having 2 extra Parser initializers just for experimental features, but SPI requires us to do it this way. The other alternative would be to make `experimentalFeatures` a `var` and just have clients set it, but I don't like that either